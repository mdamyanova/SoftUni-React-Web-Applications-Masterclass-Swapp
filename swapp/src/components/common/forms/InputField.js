import React from 'react';
import { Input } from '@rebass/forms';

const InputField = props => {
  return (
    <Input
      name={props.name}
      id={props.name}
      type={props.type}
      placeholder={props.name}
      css={`
        background-color: var(--input-background-color);
        color: var(--input-font-color);
        border-color: var(--input-border-color);
        border-radius: 5px;
      `}
      onChange={props.onChange}
    />
  );
};

export default InputField;
