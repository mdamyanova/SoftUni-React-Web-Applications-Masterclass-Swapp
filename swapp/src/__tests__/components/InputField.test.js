import React from "react";
import { shallow, mount } from "enzyme";
import InputField from "../../components/common/forms/InputField";

describe("<InputField />", () => {
  it("should match snapshot", () => {
    expect(
      shallow(<InputField name='email' type='email' placeholder='email' onChange={() => {}} />)
    ).toMatchSnapshot();
  });

  it("should have props", () => {
    const wrapper = mount(
        <InputField name='email' type='email' placeholder='email' onChange={() => {}} />
    );
    expect(wrapper.props().name).toEqual("email");
    expect(wrapper.props().type).toEqual("email");
    expect(wrapper.props().placeholder).toEqual("email");
  });
});
