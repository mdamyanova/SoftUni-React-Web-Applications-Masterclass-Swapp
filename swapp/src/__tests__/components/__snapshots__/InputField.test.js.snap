// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> should match snapshot 1`] = `
<ForwardRef
  css="
        background-color: var(--input-background-color);
        color: var(--input-font-color);
        border-color: var(--input-border-color);
        border-radius: 5px;
      "
  id="email"
  name="email"
  onChange={[Function]}
  placeholder="email"
  type="email"
/>
`;
